translator/source_code/cotrip_translator.py

```diff
@@ -0,0 +1,294 @@
+import json
+from datetime import datetime, timezone, timedelta
+import uuid
+import random
+import string
+import sys, getopt
+from jsonschema import validate
+from jsonschema import ValidationError
+import logging
+from collections import OrderedDict
+import re
+import dateutil.parser
+import translator_shared_library
+
+# Translator
+
+def main():
+    
+    inputfile, outputfile = translator_shared_library.parse_arguments(sys.argv[1:], default_output_file_name = 'cotrip_wzdx_translated_output_message.geojson')
+    if inputfile:
+
+        cotrip_obj = json.loads(open(inputfile).read())
+        wzdx_obj = wzdx_creator(cotrip_obj)
+        location_schema = '../sample files/validation_schema/wzdx_v3.0_feed.json'
+        wzdx_schema = json.loads(open(location_schema).read())
+        if not translator_shared_library.validate_wzdx(wzdx_obj, wzdx_schema):
+            print('validation error more message are printed above')
+        with open(outputfile, 'w') as fwzdx:
+            fwzdx.write(json.dumps(wzdx_obj, indent=2))
+            print('huraaah ! your wzdx message is successfully generated and located here: ' + str(outputfile))
+    else:
+        print('please specify the input file with -i')
+        print(translator_shared_library.help_string)
+
+
+    
+def wzdx_creator(message, info=None, unsupported_message_callback=None):
+    if not message:
+        return None
+   #verify info obj 
+    if not info:
+        info = translator_shared_library.initialize_info()
+    if not translator_shared_library.validate_info(info):
+        return None
+    wzd = {}
+    wzd['road_event_feed_info'] = {}
+    # hardcode
+    wzd['road_event_feed_info']['feed_info_id'] = info['feed_info_id']
```

Comment:
There seems to be a fair bit of duplicated assignments between here and the `icone_translator.py` file when creating the wzdx message. Let's abstract what we can to the shared library. For instance this entire hard-coded section of the `road_event_feed_info` assignment can be pushed to the library and likely much more.